code/game/objects/items/toys.dm

```diff
@@ -29,58 +29,72 @@
 
 /obj/item/toy/waterballoon
 	name = "water balloon"
-	desc = "A translucent balloon. There's nothing in it."
+	desc = "A translucent plastic balloon. Fun for the whole family."
 	icon = 'icons/obj/toy.dmi'
 	icon_state = "waterballoon-e"
 	item_state = "balloon-empty"
+	var/max_liquid = 10 //In case of admin fuckery or "fun" balloon subtypes
 
 /obj/item/toy/waterballoon/New()
 	. = ..()
-	create_reagents(10)
+	create_reagents(max_liquid)
+
+/obj/item/toy/waterballoon/examine(mob/user)
+	..()
+	reagents.get_examine(user)
 
 /obj/item/toy/waterballoon/attack(mob/living/carbon/human/M as mob, mob/user as mob)
 	return
 
-/obj/item/toy/waterballoon/afterattack(atom/A as mob|obj, mob/user as mob)
-	if (istype(A, /obj/structure/reagent_dispensers/watertank) && get_dist(src,A) <= 1)
-		A.reagents.trans_to(src, 10)
-		to_chat(user, "<span class = 'notice'>You fill the balloon with the contents of \the [A].</span>")
-		src.desc = "A translucent balloon with some form of liquid sloshing around in it."
-		src.update_icon()
-	return
+/obj/item/toy/waterballoon/afterattack(atom/A, mob/user, proximity_flag, click_parameters)
```

Comment:
why all this? why not just make it an opencontainer?

that would solve your examine issue AND let you fill it in any way of your choice just like any other containers, you could do away with basically all of the examine(), attackby() and afterattack()

Reply:
Well... I don't know how to is why. Oof, reagent code

Reply:
Just delete all of those 3 procs and give it the flag OPENCONTAINER. If you need help I'll be available on discord when I get home from thanksgiving.